From: Jean Delvare <jdelvare@suse.de>
Date: Thu, 6 Oct 2022 18:52:52 +0200
Subject: setup: Document the limitation of spec file support
Git-commit: b73e4afa7e4c28a981430c5b80a95254f2ed6b77
Patch-mainline: yes
References: boo#1203791

Quilt setup can't be used on arbitrary spec files. Due to the fact
that quilt only deals with patches, every other modification to the
working tree has to happen first, and the patches must be applied
last.

This is a design limitation and not a bug, but document it so that
the users are aware of it and can adjust the %prep section of their
spec file if needed.

Signed-off-by: Jean Delvare <jdelvare@suse.de>
---
 quilt/setup.in |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

--- a/quilt/setup.in
+++ b/quilt/setup.in
@@ -279,7 +279,13 @@ Initializes a source tree from an rpm sp
 
 --fast	Use the new, faster method to process the spec file. In this mode,
 	rpmbuild is told to generate a working tree directly in the target
-	directory. This is now the default.
+	directory. This is the default (since quilt version 0.67).
+
+The setup command is only guaranteed to work properly on spec files where
+applying all the patches is the last thing done in the %%prep section. This is
+a design limitation due to the fact that quilt can only operate on patches. If
+other commands in the %%prep section modify the patched files, this must
+happen first, otherwise you won't be able to push the patch series.
 "
 		exit 0
 	else
